xbmc/Application.cpp

```diff
@@ -4865,6 +4865,13 @@ bool CApplication::OnMessage(CGUIMessage& message)
       if (url.IsProtocol("plugin"))
         XFILE::CPluginDirectory::GetPluginResult(url.Get(), file);
 
+      // Don't queue if next media type is different from current one
+      if ((!file.IsVideo() && m_pPlayer->IsPlayingVideo()) ||
```

Comment:
No, I want the current player to finish playing the current media file and then switch to a different player for the next file.